### PR TITLE
feat(blog): delete blog post

### PR DESCRIPTION
**Pull request by imacpowers**
### What does this PR do?
This PR implements an endpoint for deleting a blog post from the database. The endpoint is restricted to super admin users only and provides appropriate responses for various scenarios.

- **Added DELETE /api/v1/blogs/:id Endpoint**: Allows superadmins to delete existing blog posts.
- **Authentication and Authorization**: Requires a JWT token and superadmin role for access.
- **Response Handling**: Provides appropriate responses for successful deletion, blog not found, unauthorized access, and other error scenarios.

### How should this be manually tested?
**Login Superadmin User**:
1. Make a POST request to `/api/v1/auth/login` with a superadmin user's credentials.
2. Check the response for a valid JWT token.

**Delete Blog Post**:
1. Make a DELETE request to `/api/v1/blogs/:id` with the blog ID in the path parameter and a valid JWT token in the header.
2. Verify the response confirms the blog has been deleted.

**Edge Cases**:
- **Invalid Token**: Test with an invalid token to ensure the endpoint returns a 401 Unauthorized status with the correct message.
- **Missing Token**: Test without a token to ensure the endpoint returns a 401 Unauthorized status with the correct message.
- **Non-superadmin User**: Test with a valid token of a non-superadmin user to ensure the endpoint returns a 403 Forbidden status.
- **Non-existent Blog**: Test with a non-existent blog ID to ensure the endpoint returns a 404 Not Found status.
- **Internal Server Error**: Simulate a server error to ensure the endpoint returns a 500 Internal Server Error status.

### Checklist
- [x] Implemented DELETE `/api/v1/blogs/:id` endpoint in the `blogs.controller.ts`.
- [x] Added `deleteBlogPost` method in the `blogs.service.ts`.
- [x] Modified `blogs.controller.ts` to provide appropriate responses for various cases.
- [x] Tested endpoint functionality to ensure it correctly handles valid and invalid tokens.
- [x] Added appropriate HTTP status codes and response messages for edge cases.
- [x] Added tests for this endpoint.

### Response Examples
**Success**:
```json
{
  "message": "Blog successfully deleted",
  "status_code": 202
}
```
202 response

**Forbidden**:
```json
{
  "message": "You are not authorized to perform this action",
  "status_code": 403
}
```
403 response

**Not Found**:
```json
{
  "message": "Blog with this Id does not exist",
  "status_code": 404
}
```
404 error

**Internal Server Error**:
```json
{
  "message": "Internal server error.",
  "status_code": 500
}
```

### Link to Issue
Link to this issue: #108

### CODE REVIEW

File: src/modules/blogs/blogs.controller.ts [hunk 1]
```diff
@@ -39,4 +39,19 @@ export class BlogController {
...
     };
   }
+  @Delete(':id')
+  @UseGuards(SuperAdminGuard)
+  @HttpCode(HttpStatus.ACCEPTED)
+  @ApiOperation({ summary: 'Delete a blog post' })
+  @ApiResponse({ status: 202, description: 'Blog successfully deleted.' })
+  @ApiResponse({ status: 404, description: 'Blog with the given Id does not exist.' })
+  @ApiResponse({ status: 403, description: 'You are not authorized to perform this action.' })
+  @ApiResponse({ status: 500, description: 'Internal server error.' })
```
Comment (Cobidex) [inline src/modules/blogs/blogs.controller.ts:49]: Remove this, you don't ever want your client to get this error.

File: src/modules/blogs/blogs.controller.ts [hunk 2]
```diff
@@ -39,4 +39,19 @@ export class BlogController {
...
+  async deleteBlog(@Param('id', ParseUUIDPipe) id: string): Promise<any> {
+    await this.blogService.deleteBlogPost(id);
+    return {
+      message: 'Blog successfully deleted',
```
Comment (Cobidex) [inline src/modules/blogs/blogs.controller.ts:53]: Please use constants

File: src/modules/blogs/blogs.service.ts [hunk 3]
```diff
@@ -51,6 +52,17 @@ export class BlogService {
     };
   }
 
+  async deleteBlogPost(id: string): Promise<void> {
+    const blog = await this.blogRepository.findOne({ where: { id } });
+
+    if (!blog) {
+      CustomExceptionHandler({
+        response: 'Blog with this Id does not exist',
```
Comment (Cobidex) [inline src/modules/blogs/blogs.service.ts:60]: Please use constants

**Review by Cobidex — COMMENTED** [review]
Good work, just fix comments.

File: src/modules/blogs/blogs.service.ts [hunk 4]
```diff
@@ -100,4 +100,10 @@ export class BlogService {
       created_at: updatedBlog.created_at,
     };
   }
+  async deleteBlogPost(id: string): Promise<void> {
+    const blog = await this.blogRepository.findOne({ where: { id } });
+    if (!blog) {
+      throw new CustomHttpException('Blog post with this id does not exist.', HttpStatus.NOT_FOUND);
```
Comment (Cobidex) [inline src/modules/blogs/blogs.service.ts:106]: Please use constants

**Comment by imacpowers** [comment]
@Cobidex , I've fixed all issues

File: src/modules/blogs/blogs.service.ts (hunk 4 above)
Comment (imacpowers) [reply src/modules/blogs/blogs.service.ts:106]: fixed all issues @Cobidex

File: src/modules/blogs/blogs.service.ts (hunk 3 above)
Comment (imacpowers) [reply src/modules/blogs/blogs.service.ts]: fixed

**Review by Cobidex — APPROVED** [review]
